replicated_kv_store/rpcs.go

```diff
@@ -188,7 +189,9 @@ func (node *RaftNode) AppendEntries(ctx context.Context, in *protos.AppendEntrie
 		} else {
 			node.raft_node_mutex.Unlock()
 		}
-
+		p, _ := peer.FromContext(ctx)
+		node.leader_address = p.Addr.String()
+		node.trackMessage[in.client] = node.log[int(node.commitIndex)].String()
```

Comment:
The tracked latest message for each client should be updated while an entry is being committed. In the if condition block above (if in.LeaderCommit > node.commitIndex { ...}), multiple entries may be getting committed. Adding a loop for updating trackMessage for each newly committed entry here would be appropriate. (you'll probably need to add an additional field for each log entry identifying the client that created that entry). Refer to figure 7 and it's explanation in the paper, and see if the implementation handles all the cases, including logs being modified, overwritten etc)